### ISSUE TITLE
When switching to emacs in a different space it brings the app to the current space

### ISSUE THREAD

**Issue by hisnawi** [railwaycat/homebrew-emacsmacport#182]
I am on Mojave and noticed that when trying to switch to the emacs app (using Cmd+Tab) the emacs app brings itself to the current space and it doesn't switch the view to the space emacs was originally in.

Is that on purpose? What could cause this behavior?

**Comment by iSeeU816** [comment]
@hisnawi

I know this is an old issue, but my solution might help anyone facing this issue.

This issue was driving me crazy, as I don't like windows jump around in a space
that they don't belong to. After I debugged my own Emacs configuration, I
figured out what the cause of this issue.

**TL;DR:**

1. Evaluate `(menu-bar-mode 1)`. And make sure to add it in your `init.el` file.
   - Don't forget to remove `(menu-bar-mode -1)` if it's exist in your `init.el` file.
2. If you have `(desktop-mode)` enabled, then do this:
   1. Quit Emacs.
   2. Open `.emacs.desktop` file, which is should be exist in your `.emacs.d` directory.
   3. Find and replace `(menu-bar-lines . 0)` to `(menu-bar-lines . 1)`.
   4. Save the file.
   5. Launch Emacs.

**Detailed solution:**

The cause of this issue is that in `init.el` file, I was having `(menu-bar-mode -1)`.
This is not OK for macOS as it has a separate menu bar in the top, which all
GUI (or non-background apps/tools) apps *must* (should?) have a menu bar.

Disabling the menu bar for Emacs, causing it to function in a weird way.

For that, I wrote this in my `init.el` file:

``` elisp
;; Hiding `menu-bar-mode' in macOS results in many annoying experience.
;; To name one, Emacs jumps to the current space even it is not in it.
;; Test was on Emacs Mac port 26.3 (7.8).
;;
;; NOTE: Must Evaluate this while in Emacs.
;; Then quit Emacs and change `(menu-bar-lines . 0)'
;; to `(menu-bar-lines . 1)' in `desktop' session file if it's exist.
(when (memq window-system '(mac nx x))
  (menu-bar-mode 1))
```

That's did the trick, and now I'm not having this annoying issue anymore. :)

Hope this helps!
